src/wp-includes/webfonts-api/class-wp-webfonts-registry.php

```diff
@@ -0,0 +1,241 @@
+	 *    @type string       $ascend_override         The @font-face ascend-override property.
+	 *    @type string       $descend_override        The @font-face descend-override property.
+	 * }
+	 * @return string Registration key.
+	 */
+	public function register( array $webfont ) {
+		$webfont = $this->convert_to_kebab_case( $webfont );
+
+		// Validate schema.
+		if ( ! $this->validator->is_valid_schema( $webfont ) ) {
+			return '';
+		}
+
+		$webfont = $this->validator->set_valid_properties( $webfont );
+
+		// Add to registry.
+		$registration_key = $this->generate_registration_key( $webfont );
+		if ( isset( $this->registered[ $registration_key ] ) ) {
+			return $registration_key;
+		}
+
+		$this->registered[ $registration_key ] = $webfont;
+		$this->store_for_query_by( $webfont, $registration_key );
+
+		return $registration_key;
+	}
+
+	/**
+	 * Convert snake_case keys into kebab-case.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array $webfont Webfont definition.
+	 * @return array Webfont with kebab-case properties (keys).
+	 */
+	private function convert_to_kebab_case( array $webfont ) {
+		$kebab_case = array();
+		foreach ( $webfont as $key => $value ) {
+			$converted_key                = str_replace( '_', '-', $key );
+			$kebab_case[ $converted_key ] = $value;
+		}
+
+		return $kebab_case;
+	}
+
+	/**
+	 * Store the webfont for query by request.
+	 *
+	 * This container provides a performant way to quickly query webfonts by
+	 * provider. The registration keys are stored for O(1) lookup.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array  $webfont          Webfont definition.
+	 * @param string $registration_key Webfont's registration key.
+	 */
+	private function store_for_query_by( array $webfont, $registration_key ) {
+		$provider = $webfont['provider'];
+
+		// Initialize the array if it does not exist.
+		if ( ! isset( $this->registry_by_provider[ $provider ] ) ) {
+			$this->registry_by_provider[ $provider ] = array();
+		}
+
+		$this->registry_by_provider[ $provider ][] = $registration_key;
+	}
+
+	/**
+	 * Generates the registration key.
+	 *
+	 * Format: font-family.font-style.font-weight
+	 * For example: `'open-sans.normal.400'`.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array $webfont Webfont definition.
+	 * @return string Registration key.
+	 */
+	private function generate_registration_key( array $webfont ) {
+		return sprintf(
+			'%s.%s.%s',
+			$this->convert_font_family_into_key( $webfont['font-family'] ),
+			trim( $webfont['font-style'] ),
+			trim( $webfont['font-weight'] )
+		);
+	}
+
+	/**
+	 * Converts the given font family into a key.
+	 *
+	 * For example: 'Open Sans' becomes 'open-sans'.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $font_family Font family to convert into a key.
+	 * @return string Font-family as a key.
+	 */
+	private function convert_font_family_into_key( $font_family ) {
+		if ( ! is_string( $font_family ) || '' === $font_family ) {
```

Comment:
Want to make sure to skip empty strings as well like `'      '` 

```suggestion
		if ( ! is_string( $font_family ) || '' === trim( $font_family ) ) {
```

Reply:
Hmm, relooking at this, the check here is unnecessary. Why? The value of the font-family is validated in the Validator before this method is called.

Reply:
Removed the checks in this commit https://github.com/WordPress/wordpress-develop/pull/1736/commits/f5bc3064eb75103e2a411da582673513ce8e1aa8.